### DISCUSSION TITLE
Grafana Charge Dashboard kw price

### DISCUSSION THREAD

**Discussion by Shadesh** [teslamate-org/teslamate#3536] [Q&A] [answered]
As far as i can remember, older Teslamate versions had the kw price calculated in the Grafana Charges Dashboard. No it only shows Duration, Cost and Added. But not longer the price per kw. Is there some way to get this back?

**Comment by DrMichael** [comment, marked as answer]
I added it in [PR-3537](https://github.com/teslamate-org/teslamate/pull/3537)
I should say, that it is not completely accurate.

**Comment by DrMichael** [comment]
Can someone tell, why it is in `master` but not in `latest` and `edge`?

**Reply by cwanja** [reply]
For `latest`, it is not in a "release" yet. Although, I would think `edge` includes it. But not familiar enough with GitHub / Docker releases to be definitive in that. @JakobLichterfeld @brianmay do either of you know regarding `edge` "versions" in Docker.

**Reply by JakobLichterfeld** [reply]
Latest is the latested release version, so actual v1.28.2 tag branch.

Edge is build each Sunday at 3 o'clock UTC, see https://github.com/teslamate-org/teslamate/blob/de8227ce0b9fcb78532d68a0da206bf2e192c31f/.github/workflows/buildx.yml#L6

To test the PR, there is a dedicated GHCR Image, see my comment here https://github.com/teslamate-org/teslamate/pull/3537#issuecomment-1838686816

**Reply by DrMichael** [reply]
I found the cause: calculation of cost per kWh was missing in master.
Added it back in https://github.com/teslamate-org/teslamate/pull/3544

**Comment by vincep5** [comment]
Also, in the Charges dashboard the last column Odometer is showing me km instead of mi

**Reply by cwanja** [reply]
Fixed in #3534 which is merged. Look out for next release.
